onedal/cluster/kmeans.py

```diff
@@ -261,18 +263,33 @@ def _fit_backend(
         )
 
     def _fit(self, X, module, queue=None):
-        policy = self._get_policy(queue, X)
         is_csr = _is_csr(X)
-        X = _check_array(
-            X, dtype=[np.float64, np.float32], accept_sparse="csr", force_all_finite=False
-        )
+
+        use_raw_input = _get_config().get("use_raw_input") is True
+        if use_raw_input and _get_sycl_namespace(X)[0] is not None:
```

Comment:
```suggestion
        if use_raw_input and sua_iface is not None:
```
+ move line 284 above this